src/client/bots/GreedyBot.cpp

```diff
@@ -0,0 +1,25 @@
+#include "GreedyBot.hpp"
+
+GreedyBot::GreedyBot() {};
+
+std::pair<InputDef::ReliableInput, InputDef::UnreliableInput>
+GreedyBot::getMove() {
+    // run bot getMove(gameState) function
+    // return move in m_reliableInput and m_unreliableInput
+    srand(time(0));
+    bool rand_bool = bool(rand() % 2);
+    m_reliableInput = {
+        .toggle_ungroup = false,
+        .toggle_joinable = false,
+    };
+    m_unreliableInput = {
+        .toggle_up = false,
+        .toggle_down = false,
+        .toggle_right = bool(rand() % 2),
+        .toggle_left = false,
+        .toggle_stop = false,
+    };
+
+    return std::pair<InputDef::ReliableInput, InputDef::UnreliableInput>(m_reliableInput,
```

Comment:
Maybe the bot logic could produce a struct of more intuitive inputs and then we can have a util function all bots share that converts this to to `ReliableInput`/`UnreliableInput`.

Reply:
it does feel like this is a violation of the abstraction. like a bot shouldn't care about the reliable/unreliability of the input, just cares about a "move"

not high pri to fix for now, but worth keeping in mind yeah